### PR TITLE
Update releases.md

### PR DESCRIPTION
**Pull request by JonathanGiles**

### CODE REVIEW

File: docs/policies/releases.md [hunk 1]
```diff
@@ -199,16 +199,16 @@ Preview .NET packages will be published to NuGet with the pre-release designatio
 
 Maven supports the [convention](https://cwiki.apache.org/confluence/display/MAVENOLD/Versioning) `MAJOR.MINOR.PATCH-QUALIFIER`. The preferred format for version numbers is:
 
-- `X.Y.Z-dev.YYYYMMDD.r` (`r` is based on the number of builds performed on the given day)
-- `X.Y.Z-preview.N`
+- `X.Y.Z-dev-YYYYMMDD-r` (`r` is based on the number of builds performed on the given day)
```
Comment (weshaggard) [inline docs/policies/releases.md]: Can you give more context here? Does Maven that support semver 2 correctly?

Comment (weshaggard) [reply docs/policies/releases.md]: We should definitely get clarity on this and add a link to whatever documentation we use to make a decision here.

Comment (JonathanGiles) [reply docs/policies/releases.md]: https://books.sonatype.com/mvnref-book/reference/pom-relationships-sect-pom-syntax.html is a good reference link. It also has an interesting point to left-pad the preview numbers to do proper string comparison.

Comment (JonathanGiles) [reply docs/policies/releases.md]: https://blog.soebes.de/blog/2017/02/04/apache-maven-how-version-comparison-works/ is also interesting

Comment (JonathanGiles) [reply docs/policies/releases.md]: And https://stackoverflow.com/questions/13004443/how-does-maven-sort-version-numbers

The only difficult thing is reconciling the sometimes conflicting advice. I think we might almost need to do some tests locally to ensure we are properly conforming. Once we are happy we will also need to mass update version strings.

**Comment by weshaggard** [comment]
@JonathanGiles I hope you don't mind I took over your PR and updated based on my findings.

After reading through lots of blogs and docs I was still not able to figure out the versioning rules, particularly around pre-releases/qualifiers. So I went to reading the code and even then I couldn't figure out the exact orders so I actually cloned maven and started running some code and added some tests to verify sorting of a bunch of qualifier combinations with the full results at https://github.com/weshaggard/maven/blob/AddVersioningTests/maven-artifact/src/test/java/org/apache/maven/artifact/versioning/ComparableVersionTest.java#L270-L398.

To boil it down to just the things we were considering here is what the sort order looks like:
```
1.0.0-beta.1
1.0.0
1.0.0-dev.201911056.1
1.0.0-preview.1
```

Maven has a fixed set of [qualifiers](https://github.com/apache/maven/blob/master/maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java#L356-L357), which includes the empty/GA qualifier in its list, and any qualifier that isn't part of that set sorts higher then anyone in that set, again including GA. The code that does that sorting is at https://github.com/apache/maven/blob/master/maven-artifact/src/main/java/org/apache/maven/artifact/versioning/ComparableVersion.java#L423-L428. 

It is for that reason we have to use one of their predefined qualifiers for anything we publish to maven central in order for them to sort correctly. I've chosen `beta` to be consistent with other languages that have a similar issue. It is also worth nothing it doesn't matter if we use `-` or `.` it will correctly sort the numbers numerically, in fact we don't have to use any separator but I chose to use `.` to be more consistent with our others. 

As for our `dev` builds we could use `alpha` as a qualifier but given we plan to never publish them to maven I thought sticking with `dev` to help easily identify the things we only want for private testing is a good way go. It also allows us to remain consistent with the other languages.

**Comment by hemanttanwar** [comment]
I have compared few example version using this 'ComparableVersion'  and here is result 
https://maven.apache.org/ref/3.3.1/maven-artifact/apidocs/org/apache/maven/artifact/versioning/ComparableVersion.html 
Like @weshaggard  suggested ,  We can go with #3 and #4 approach . I would prefer #4 

1. ---------------------------------
   1.0.0-preview.2 < 1.0.0-preview.3
2. ---------------------------------
   1.0.0-preview.2 > 1.0.0  ( This is the problem we have today)
3. ---------------------------------
   1.6.0 > 1.6.0.m1
4. ---------------------------------
   1.0.0-beta.2 < 1.0.0-beta.3
---------------------------------
   1.0.0-beta.2 < 1.0.0

@JonathanGiles

**Comment by JonathanGiles** [comment]
Thanks for taking this on @weshaggard and @JimSuplizio. If `beta` is the right choice to be consistent with other languages, go for it. I would just register an interest in considering the `milestone` or `rc` options as possible contenders too - but I agree that consistency with other languages would be nice.

**Pull request merged**